srcpkgs/pinephone-uboot/files/kernel.d/uboot

```diff
@@ -0,0 +1,53 @@
+#!/bin/sh
+
+kver=${2}
+uuid=$(cat /etc/fstab | awk '$2 == "/" { print $1 }' | sed 's/^UUID=//')
+dev=$(blkid -U "${uuid}")
+partuuid=$(blkid -o value -s PARTUUID "${dev}")
+bootpart=$(df -P /boot | tail -1 | awk '{ print $6 }')
```

Comment:
Sorry, this part is a bit complicated.
And it doesn't account for multiple layer fs, (`lvm`, `luks`), right?

*I think* it's better to ask user specify `bootargs` directly in a configuration file.
It seems like other bootloader don't try to guess this value. :-)

I don't have PinePhone, so I could mis-understand everything I said, though :)

Reply:
Even if we agree of this logic, is it better to use:
```
df -P / | awk 'NR == 2 {print $1}'
```
instead of relying on forcing on `UUID=` on `/etc/fstab`?

Reply:
I was thinking of having something like a `/etc/default/cmdline` for specifying boot args, but I didn't want to stray too far from the pinebookpro template. I agree that it would be better to make this more end-user friendly, so I'm willing to try something else.

As for the complicated part, I 100% agree. I tried something a little less hacky by using blkid's `eval` output mode, which just spitss out conf style variable assignments eg `PARTUUID=deadbeef` that can be sourced by sh.

I personally think the best approach for flexibility reasons would be to source cmdline arguments from `/etc/default/cmdline` and have this configure hook fill it into the boot script. It sits inline with what `grub-mkconfig` does with `/etc/default/grub`, and `void-mklive` could just append `root=...` when creating base images.

Reply:
The reconfigure hook now reads settings from `/etc/default/pinephone-uboot-config` to determine the cmdline and toggle cmdline parameters for the volume rocker state.